### PR TITLE
Fix plotting vector fields

### PR DESCRIPTION
**Pull request by Yuriyzabegaev**
## Proposed changes

The function `pp.plot_grid` now doesn't work if you pass `mdg` and `vector_value: str`. The internal function `_quiver` expects the array of shape (3 x n), but the state contains 1D arrays. However, this visualization works if you pass not `mdg`, but `grid`, and `vector_value: ndarray`. In this case, the user ensures that he provides the array of shape (3 x n). So we have 2 cases of behaviour:

1. `mdg` and `vector_value: str` - fails due to an error. `mdg` and `vector_value: ndarray` is not allowed according to the docstring.
2. `grid` and `vector_value: ndarray` - works.

All the examples use the second case. I made a minor fix so the first case is not failing, and verified that it doesn't break the second case.

I provided the tests for both cases. Also, I provided a test for the function `save_img` as @jhabriel suggested. The tests cover three setups: simplex 2D grid, cartesian 2D grid and cartesian 3D grid. The vector variable is stored at the cell centres everywhere, but I hope it must not affect the behaviour of the plotting function significantly.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply_

- [ ] Minor change (e.g., dependency bumps, broken links, etc).
- [x] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code, etc).
- [ ] Other: 

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.	
- [ ] Static typing is included in the update.
- [x] This PR does not duplicated existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).

### CODE REVIEW

File: src/porepy/viz/plot_grid.py [hunk 1]
```diff
@@ -276,10 +276,19 @@ def plot_mdg(
         # Adjust rgb colors depending on the subdomain ordering
         kwargs["rgb"] = np.divide(kwargs.get("rgb", [1, 0, 0]), sd.id + 1)
         # Plot the subdomain and data
+
+        vector_value_array = sd_data.get(pp.STATE, {}).get(vector_value, None)
+        if vector_value_array is not None:
+            # The further algorithm requires the vector_value array of shape (3 x n). but
+            # We have the 1D data array. Thus, we fill the remaining dimensions with zeros.
+            vector_value_array = vector_value_array.reshape(sd.dim, -1, order="F")
```
Comment (IvarStefansson) [inline src/porepy/viz/plot_grid.py:284]: I have a feeling shape should be (mdg.dim_max(), -1) or perhaps rather (-1, sd.num_cells)

File: src/porepy/viz/plot_grid.py [hunk 2]
```diff
@@ -276,10 +276,19 @@ def plot_mdg(
         # Adjust rgb colors depending on the subdomain ordering
         kwargs["rgb"] = np.divide(kwargs.get("rgb", [1, 0, 0]), sd.id + 1)
         # Plot the subdomain and data
+
+        vector_value_array = sd_data.get(pp.STATE, {}).get(vector_value, None)
+        if vector_value_array is not None:
+            # The further algorithm requires the vector_value array of shape (3 x n). but
```
Comment (IvarStefansson) [inline src/porepy/viz/plot_grid.py:282]:
```suggestion
            # The further algorithm requires the vector_value array of shape (3 x n). But
```

File: tests/unit/test_plotting.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+)
+def mdg(request):
+    """Helper for initialization of the parametrized mdgs"""
+    mdg_ = request.param[0]
+    _initialize_mdg(mdg_)
+    return mdg_
+
+
+def test_plot_grid_mdg(mdg):
+    """Tests that no error is risen if we plot mdg and provide variable names."""
```
Comment (IvarStefansson) [inline tests/unit/test_plotting.py:32]:
```suggestion
    """Tests that no error is raised if we plot mdg and provide variable names."""
```

**Review by IvarStefansson — COMMENTED** [review]
Nice work! One substantial comment, rest is cosmetics.

File: src/porepy/viz/plot_grid.py (hunk 1 above)
Comment (Yuriyzabegaev) [reply src/porepy/viz/plot_grid.py:284]: (mdg.dim_max(), -1) - what if the domain is 3D, the subdomain and the data are 2D?
(-1, sd.num_cells) - the data can be defined at the edges.

Why don't you prefer my choice - (sd.dim, -1)?

Comment (IvarStefansson) [reply src/porepy/viz/plot_grid.py:284]: The vector is usually nd. E.g. traction on a fracture is a nd-vector, even though the fracture is 2d.
Does the vector plotting work for faces/edges?

Comment (keileg) [reply src/porepy/viz/plot_grid.py:284]: Alternatively, the dimension of the vector field can be made an optional input argument, with the default being (-1, sd.num_cells).

Comment (IvarStefansson) [reply src/porepy/viz/plot_grid.py:284]: Yes, if there's a (not too exotic) use case for that.

Comment (Yuriyzabegaev) [reply src/porepy/viz/plot_grid.py:284]: > The vector is usually nd. E.g. traction on a fracture is a nd-vector, even though the fracture is 2d. Does the vector plotting work for faces/edges?

@IvarStefansson I see your point. I believe, (mdg.dim_max(), -1) must cover all possible cases. The vector plotting was intended to work with faces/edges, see: https://github.com/pmgbergen/porepy/blob/d4cc3adcbecf5401a5b2c2dcc4398e586cf5d9ff/src/porepy/viz/plot_grid.py#L395

@keileg probably, that will be superfluous. Taking mdg.dim_max() will always allow us to understand if the vector is placed in cell centres or edges etc. Or maybe there is some exotic use case I don't see now.

Comment (IvarStefansson) [reply src/porepy/viz/plot_grid.py]: I agree. If we ever need it we can always add a vector_dim-argument later.

**Pull request merged**